kalite/distributed/static/js/distributed/exercises/software-keyboard.js

```diff
@@ -0,0 +1,150 @@
+var BaseView = require("base/baseview");
+var Handlebars = require("base/handlebars");
+var _ = require("underscore");
+var $ = require("base/jQuery");
+
+var SoftwareKeyboardView = BaseView.extend({
+
+    template: require("./hbtemplates/software-keyboard.handlebars"),
+
+    events: {
+       "click button.key" : "key_pressed",
+       "click button#show-keyboard": "toggle_keypad",
+       "keypress button": "catch_keypress"
+     },
+
+
+    keys: [
+        [ {display: "1", value: "1"}, {display: "2", value: "2"}, {display: "3", value: "3"} ],
+        [ {display: "4", value: "4"}, {display: "5", value: "5"}, {display: "6", value: "6"} ],
+        [ {display: "7", value: "7"}, {display: "8", value: "8"}, {display: "9", value: "9"} ],
+        [ {display: "(", value: "("}, {display: "0", value: "0"}, {display: ")", value: ")"} ],
+        [ {display: "/", value: "/"}, {display: ".", value: "."}, {display: "-", value: "-"} ],
+        [ {display: "c", value: "c"}, {display: "␣", value: " "}, {display: "Del", value: "Del"} ]
+    ],
+
+    initialize: function () {
+
+        _.bindAll(this, "set_input", "toggle_keypad", "key_pressed", "render", "hide", "show", "catch_keypress");
+
+        this.touch = Modernizr.touch;
+        this.enabled = true;
+        this.render();
+
+    },
+
+    set_input: function(el) {
+        this.inputs = $(el)
+            .prop("readonly", this.enabled)
+            .css("-webkit-tap-highlight-color", "rgba(0, 0, 0, 0)");
+    },
+
+    toggle_keypad: function() {
+        var self = this;
```

Comment:
self is defined here for use later on, when the scope of 'this' might make a difference to what 'this' refers to.

Hence, inside a callback function, self is used, rather than 'this' - but everywhere else, 'this' is fine.